modules/core/map/js/farmOS.map.behaviors.input.js

```diff
@@ -1,11 +1,56 @@
 (function () {
   farmOS.map.behaviors.input = {
     attach: function (instance) {
+
+      // Get the data input form element.
+      var input = instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]')
+
+      // When features change in the edit layer, write WKT to the input form element.
       instance.editAttached && instance.editAttached.then(() => {
         instance.edit.wktOn('featurechange', function(wkt) {
-          instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]').value = wkt;
+          input.value = wkt;
         });
       });
+
+      // Run a handleInput() callback when input changes and attempt to
+      // import the new WKT.
+      input.oninput = handleInput;
+      function handleInput(e) {
+        if (input.value) {
+
+          // Add a new temporary invisible WKT layer.
+          var layer = instance.addLayer('wkt', {
+            title: 'WKT',
+            wkt: input.value,
+            visible: false,
+          });
```

Comment:
What happens if the WKT is invalid? Currently this is handled server-side and an error is surfaced, but this would silently fail as-written right?

I'm guessing we need to catch the errors and surface them on/around the input text field... possibly using `input.setCustomValidity(message)`. See https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#validating_forms_using_javascript

Reply:
It silently fails currently.

Can we tackle this in a followup, and just let Drupal surface any errors on form save for now?

Reply:
Seems reasonable